packages/nimble-components/src/menu-button/tests/menu-button.spec.ts

```diff
@@ -190,14 +186,14 @@ describe('MenuButton', () => {
         it('should not open menu when the toggle button is clicked if the element is disabled', async () => {
             element.disabled = true;
             await connect();
-            element.toggleButton!.control.click();
+            pageObject.clickMenuButton();
             expect(element.open).toBeFalse();
         });
 
         it('should close menu when toggle button is clicked while the menu is open', async () => {
             element.open = true;
             await connect();
-            element.toggleButton!.control.click();
+            pageObject.clickMenuButton();
             expect(element.open).toBeFalse();
```

Comment:
Since they're in the page object API, we should presumably be using those instead:
```ts
    await pageObject.openMenu();
    pageObject.clickMenuButton();
    expect(pageObject.isOpen()).toBeFalse();
```

This applies to other places in this file, too.

Reply:
I could update the tests so that none (or very few) of them programmatically open/close the menu button setting `element.open` explicitly, but I'm not sure I see the advantage of that. Could you elaborate on why you think it would be better to use `await pageObject.openMenu()` rather than setting `element.open = true`?

Reply:
My thinking was that you've created a page object function that abstracts away the details of opening the menu, so that would be the preferred way to open it, but on second thought, `open` is part of the menu button API, so I guess that's just as valid. If that's the case, then the primary purpose of the page object function must be to allow waiting for the menu to actually be open. It's not clear to me why we sometimes need that, and other times can just call `open = true`. 

Also, I noticed that the page object's `openMenu()` opens via click. Should it instead use `open = true`, since that's what the old definitions of `openMenu()` were doing? If it is relevant that it opens via a click, maybe it should be named `clickMenuButtonAndWait()`.

Reply:
My thoughts are:
- The page object should open the menu in an interactive way (through click or key events) since presumably almost any client test is trying to test how a user would interact with the page rather than via programmatic means
- In the end, a client calling `pageObject.openMenu()` doesn't care how the menu is opened. If they do care, they should call some other function such as `clickMenuButton` or `pressEnterOnMenuButton` so that they are explicitly controlling the precise interaction.
- These tests can be a little more efficient by setting `open = true` before calling `await connect()` with no further need to wait for the menu to render as open. Otherwise, they'd need to `await connect()` and then `await pageObject.openMenu()`.